ale_linters/json/eslint.vim

```diff
@@ -0,0 +1,16 @@
+" Author: João Pesce <joao@pesce.cc>
+" Description: eslint for JSON files.
+"
+" Requires eslint-plugin-jsonc or a similar plugin to work
+"
+" Uses the same funtcions as ale_linters/javascript/eslint.vim by w0rp
+" <devw0rp@gmail.com>
+
+call ale#linter#Define('json', {
+\   'name': 'eslint',
+\   'output_stream': 'both',
+\   'executable': function('ale#handlers#eslint#GetExecutable'),
```

Comment:
Would be better to create a different GetExecutable function that somehow checks if the plugin for JSON linting is available. If not available return an empty string so the linter is not run for these filetypes if the plugin is not installed.

To run commands (e.g. for checking if the plugin is present) without freezing vim/neovim make sure to use the ale#comand#Run() function that works in background. Check any linter that uses this function (e.g. eclipselsp.vim) to determine the version of the tools used  for examples.

Reply:
Hi @hsanson, thanks for reviewing this PR!  I'm not the author, but am excited to see this added.

> Would be better to create a different GetExecutable function that somehow checks if the plugin for JSON linting is available.

Although I agree with this idea, I think it's a bit unreasonable to require it as part of this PR, since neither `graphql/eslint` nor `typescript/eslint` implement such checks as far as I can see.  Also, doing the check in `GetExecutable()` would cause eslint to be invoked twice on every call, unless some (tricky) caching logic were added, correct?

Reply:
Hmmm, to me it makes sense what @kevinoid said. 

Anyway, I was trying to figure out how to check if a plugin is installed or not, but I haven't found a solid approach:

Using `npm list` or something similar requires the user to be actually using `npm` (we would also need to check with `yarn`). Plus, even if the plugin is installed, it may not be properly configured in `.eslintrc` (not using `jsonc-eslint-parser` as the parser for JSON* files for example).

`eslint --print-info ficticiousfile.json` might work, but I found out there are multiple eslint plugins for JSON (for example ([eslint-plugin-json](https://www.npmjs.com/package/eslint-plugin-json) and [eslint-plugin-jsonc](https://www.npmjs.com/package/eslint-plugin-jsonc) both work with JSON).

I think the ALE commands (e.g. `:ALEInfo`) gives the user enough info to figure out what's going on and decide for themselves.

Let us know what you think @hsanson. 🐸

Reply:
> Although I agree with this idea, I think it's a bit unreasonable to require it as part of this PR, since neither graphql/eslint nor typescript/eslint implement such checks as far as I can see.

Indeed, this is just a recommendation, not a requirement. I do not use eslint myself so I have no strong opinions on the matter.  If this works for those that do use the tool I am fine with it.

> Anyway, I was trying to figure out how to check if a plugin is installed or not, but I haven't found a solid approach

Thanks for taking the time to investigate. If it is too complicated to do then better leave it as it is. Simple is best.
We can improve on this in the future if it becomes an issue .